scripts/commands/updateApiDocs.ts

```diff
@@ -370,15 +369,58 @@ function urlToPath(url: string) {
   return `${getRoot()}/docs${url}.md`;
 }
 
-async function createHistoricalFolder(pkgName: string, outputDir: string) {
-  mkdirp(outputDir);
+/**
+ * Adds a new entry for the release notes of the current API version to the _toc.json
+ * of all historical API versions.
+ */
+async function updateHistoricalTocFiles(pkg: Pkg) {
+  console.log("Updating _toc.json files for the historical versions");
+
+  const historicalFolders = (
+    await readdir(`${getRoot()}/docs/api/${pkg.name}`, { withFileTypes: true })
+  ).filter((file) => file.isDirectory() && file.name.match(/[0-9].*/));
+
+  for (let folder of historicalFolders) {
+    let tocFile = await readFile(
+      `${getRoot()}/docs/api/${pkg.name}/${folder.name}/_toc.json`,
+      {
+        encoding: "utf8",
+      },
+    );
+
+    let jsonData = JSON.parse(tocFile);
+
+    // Add the new version if necessary
+    for (let child of jsonData.children) {
+      if (child.title == "Release notes") {
+        addNewReleaseNoteToc(child, pkg.versionWithoutPatch);
+      }
+    }
+
+    await writeFile(
+      `${getRoot()}/docs/api/${pkg.name}/${folder.name}/_toc.json`,
+      JSON.stringify(jsonData, null, 2) + "\n",
+    );
+  }
+}
 
-  // All projects have a single release notes file except Qiskit, which has a
-  // subfolder to store the release notes for each historical version.
+/**
+ * Adds a new entry for the current API version to the JSON list of release notes
+ */
+function addNewReleaseNoteToc(releaseNotesNode: any, newVersion: string) {
+  // We need to duplicate the first entry when we converted the current version
+  // to historical to ensure having that version at the top.
```

Comment:
Maybe add an assertion that the new version is > than the old version? Because otherwise the ordering will be wrong without people realizing.

Reply:
Good idea! However, I deleted that piece of code because the conditional is no longer necessary. That is because we don't have the duplicated entries on each historical `_toc.json` file anymore :+1: